src/main/java/emu/grasscutter/game/ability/AbilityManager.java

```diff
@@ -364,6 +366,23 @@ private void handleModifierChange(AbilityInvokeEntry invoke) throws Exception {
         var modChange = AbilityMetaModifierChange.parseFrom(invoke.getAbilityData());
         var head = invoke.getHead();
 
+        var entity = this.player.getScene().getEntityById(invoke.getEntityId());
+        if (entity == null) {
+            Grasscutter.getLogger().debug("Entity not found: {}", invoke.getEntityId());
+            entity = this.player.getScene().getRemovedEntityById(invoke.getEntityId());
+            if (entity == null) return;
+            Grasscutter.getLogger().debug("Using removed Entity: {}", invoke.getEntityId());
+        }
+
+        // Destroying rocks
+        if (entity instanceof EntityGadget targetGadget
+            && targetGadget.getContent() instanceof GadgetGatherObject gatherObject) {
```

Comment:
Hmm, looks kinda weird to put this hard coded in here, since its not directly part of the invoke entry actions.
Maybe it would be better to put this as a die action if there is no gadget lua or ability for this.
Might maybe be good to talk with @StartForKiller about this.

Reply:
applying this to the resources should properly fix the problem with the drops:
https://github.com/Anime-Game-Servers/CustomGCResources/pull/7